File: mergeSort.py
def print_array(arr):
    for num in arr:
        print(num, end=" ")
    print()

def merge(arr, mid, low, high):
    i = low
    j = mid + 1
    k = low
    B = [0] * 100

    while i <= mid and j <= high:
        if arr[i] < arr[j]:
            B[k] = arr[i]
            i += 1
            k += 1
        else:
            B[k] = arr[j]
            j += 1
            k += 1

    while i <= mid:
        B[k] = arr[i]
        k += 1
        i += 1

    while j <= high:
        B[k] = arr[j]
        k += 1
        j += 1

    for i in range(low, high + 1):
        arr[i] = B[i]

def merge_sort(arr, low, high):
    if low < high:
        mid = (low + high) // 2
        merge_sort(arr, low, mid)
        merge_sort(arr, mid + 1, high)
        merge(arr, mid, low, high)

if __name__ == "__main__":
    n = int(input("Enter the number of elements: "))
    arr = []

    print("Enter the elements of the array:")
    for _ in range(n):
        element = int(input())
        arr.append(element)

    print("Array before sorting:")
    print_array(arr)

    merge_sort(arr, 0, n - 1)

    print("Array after sorting:")
    print_array(arr)

File: mergeSort.java
import java.util.*;

public class mergeSort {
    public static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void merge(int arr[], int mid, int low, int high) {
        int i, j, k;
        i = low;
        j = mid + 1;
        k = low;
        Map<Integer, Integer> B = new HashMap<>();

        while (i <= mid && j <= high) {
            if (arr[i] < arr[j]) {
                B.put(k, arr[i]);
                i++;
                k++;
            } else {
                B.put(k, arr[j]);
                j++;
                k++;
            }
        }

        while (i <= mid) {
            B.put(k, arr[i]);
            k++;
            i++;
        }

        while (j <= high) {
            B.put(k, arr[j]);
            k++;
            j++;
        }

        for (int idx = low; idx <= high; idx++) {
            arr[idx] = B.get(idx);
        }
    }

    public static void mergeSort(int arr[], int low, int high) {
        if (low < high) {
            int mid = (low + high) / 2;
            mergeSort(arr, low, mid);
            mergeSort(arr, mid + 1, high);
            merge(arr, mid, low, high);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];

        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.println("Array before sorting:");
        printArray(arr);

        mergeSort(arr, 0, n - 1);

        System.out.println("Array after sorting:");
        printArray(arr);

        scanner.close();
    }
}
